phoenix-core/src/main/java/org/apache/phoenix/coprocessor/PhoenixTTLRegionObserver.java

```diff
@@ -204,17 +213,22 @@ public PhoenixTTLRegionScanner(RegionCoprocessorEnvironment env, Scan scan,
 
         private boolean doNext(List<Cell> result, boolean raw) throws IOException {
             try {
+                long startTime = EnvironmentEdgeManager.currentTimeMillis();
                 boolean hasMore;
                 do {
                     hasMore = raw ? scanner.nextRaw(result) : scanner.next(result);
                     if (result.isEmpty()) {
                         break;
                     }
+                    if (isDummy(result)) {
+                        return true;
+                    }
                     numRowsScanned++;
                     if (maskIfExpired && checkRowNotExpired(result)) {
                         break;
                     }
 
+                    //TODO 6211 The following logic does not sound correct
```

Comment:
@jpisaac can you please comment on this?

Reply:
@gjacoby126 @kadirozde if this comment was related to the test failures, then as I indicated we have a fix for it. The test failures were happening when deleting from local indexes